### ISSUE TITLE
Two-Week Limit on Planner View

### ISSUE THREAD

**Issue by iwannet** [super-productivity/super-productivity#3702]
## Problem Statement
Currently, the planner view only allows you to view and add tasks for the next two weeks, which limits its effectiveness for long-term planning.

## :grey_question: Possible Solution
To enhance long-term planning, i would suggest to allow users to view and add tasks for an extended timeframe (to the planner view), potentially even indefinitely.

## :arrow_heading_up: Describe alternatives you've considered
Using another planner like tweek.so but that one isn't ffoss

**Comment by johannesjo** [comment]
Hey @panoramix360 !  I think, I'd prefer lazy loading or a button at the end of all loaded days. 

> * If the user loads a lot of days, I don't know if this could overload the UI.

Yes and it could potentially crash the app :D  Maybe we can add a 2 Month limit for now?

**Comment by iwannet** [comment]
I propose adding an arrow button that allows users to navigate between weeks. This would enable users to jump directly to the next week. (also eliminating the current two-week sliding feature). It will also prevent users from loading months of data at once, reducing the risk of the app crashing.

**Comment by iwannet** [comment]
> Provide a button for the user to add a task for a custom date, and then render this weeks on the screen as needed (this one probably needs a concept UI)

I like this idea to, but it might be more challenging to implement effectively with a good GUI, and if not done right, it will probably make it more confusing to the user
### Pros
- Probably uses less data
- User has a clean overview when **viewing** tasks
### Cons
- User has a less clear overview when **adding/planning** tasks

**Comment by johannesjo** [comment]
@iwannet normally this would have been my go to solution. But I see two problems with that approach:
1. You can't drag and drop tasks between a Friday and a Monday of the next week
2. On many screens and window sizes, there is not enough space for 7 days, so you would have to scroll and click if .

On the other hand I also like using the common time frame of Monday - Sunday (and maybe next Monday for the current week).

In conclusion all approaches have there advantages and disadvantages. I think I would be fine with either one of them.

@Jagdfalke what would you prefer?

**Comment by Jagdfalke** [comment]
I'm not sure whether I should have a say in this. 

To me, lazy loading sound fine. Limiting the overall timeframe to 2 or maybe even 3 months should be viable. I doubt many will actually drag and drop tasks 3 or 4 weeks in the future.

**Comment by iwannet** [comment]
still relevant

**Comment by warreth** [comment]
Maybe this can be accomplished by allowing users to click on days in the Schedule overview to edit a day/add new tasks to that day.
Then you could allow users to not only view the current month, but also the next month?
